feat(container): document new way to load renderers (#8428)
withastro/docs: src/content/docs/en/reference/container-reference.mdx

#### `renderers` option

**Type:** `SSRLoadedRendere[]`

> A list of client renderers required by the component. Use this if your `.astro` component renders any [UI framework components](/en/guides/framework-components/) using an official Astro integration (e.g. React, Vue, etc.). For each framework rendered, you must provide an object stating the integration package `name`, as well as both its client and server rendering script.

Reviewer: We've now lost the example that the second sentence describes, so people will have no reference how to do it the other way in "non-viteful environments". Did you mean to leave that earlier paragraph as is with no changes? Is that previous way still OK? If so, when/for what things?

Do we want to instead "switch" to only talking about the new way to load renderers, rather than make this an alternative?

> In a vite-full environment, renderers should be loaded by using the function `loadRenderers`, from the virtual module `astro:container`.

Reviewer: I'm not sure "a vite-full environment" is clear enough to be helpful. I'm not sure what specific conditions this does or doesn't include. Can you be more descriptive here? (Even to me, then I can put it into words if you like!)

The following example provides the necessary object to render an Astro component that renders a React component: 

```js

> import { getContainerRenderer } from "@astrojs/react";

Reviewer: ```suggestion
import { getContainerRenderer } from "@astrojs/react";
```
I'm assuming this is what you import from each different renderer integration. What happens when you have multiple to render (e.g. from `@astrojs/vue` or `@astrojs/svelte` at the same time? Are they all called exactly the same thing, and is that OK? (No naming collision?)

We do say above you need to do this "for each framework rendered" and I'm not immediately sure what that would look like if you tried to render two frameworks. Maybe the example below could be updated to show multiple? (I think then, it will be pretty easy for the reader to visualize what only one would look like.)

#### `renderers` option

**Type:** `SSRLoadedRenderer[]`

> A list of client renderers required by the component. Use this if your `.astro` component renders any [UI framework components](/en/guides/framework-components/) using an official Astro integration (e.g. React, Vue, etc.). For each framework rendered, you must use function `getContainerRenderer` that can be imported from the renderer library.

Reviewer: ```suggestion
A list of loaded client renderers required by the component. Use this if your `.astro` component renders any [UI framework components](/en/guides/framework-components/) or MDX using an official Astro integration (e.g. React, Vue, etc.).

For each official Astro integration, use its `getContainerRenderer()` helper function that can be imported from package to expose its client and server rendering scripts. These are available for `@astrojs/react`, `@astrojs/preact`, `@astrojs/solid-js`, `@astrojs/svelte`, `@astrojs/vue`, `@astrojs/lit`, and `@astrojs/mdx`.

For renderer packages outside the `@astrojs` npm org, look in their documentation for `getContainerRenderer()` or a similar function provided.
```
Tried to incorporate the note here!

Also, I added "a list of LOADED client renderers" to further emphasize the `loadRenderers()` function use, and because the type has changed. I thought maybe this is nice for consistency? Up to you if you think that is helpful!

> When using `vite` (`vitest`, Astro integrations, etc.), the renderers should be loaded by using the function `loadRenderers`, from the virtual module `astro:container`. Outside `vite`, you'll have to load the renderers manually.

Reviewer: ```suggestion
When using `vite` (`vitest`, Astro integrations, etc.), the renderers are loaded with the function `loadRenderers()` from the virtual module `astro:container`. Outside `vite`, you'll have to load the renderers manually.
```

The following example provides the necessary object to render an Astro component that renders a React component and a Svelte component: 

```js
import { getContainerRenderer as reactContainerRenderer } from "@astrojs/react";
import { getContainerRenderer as svelteContainerRenderer } from "@astrojs/svelte";
import { loadRenderers } from "astro:container";

const renderers = loadRenderers([reactContainerRenderer(), svelteContainerRenderer()]);
const container = await experimental_AstroContainer.create({
    renderers
})
const result = await container.renderToString(ReactWrapper);
```

> :::note
> The function `getContainerRenderer` is only available for renderers inside the `@astrojs/` npm org. Renderers outside this org will provide a similar/same function.
> :::
> 

Reviewer: ```suggestion
```
We try to avoid notes when we can, especially if we can work the text into a normal paragraph. I tried to do that above. See what you think!
